src/commands/instances/group/GroupRecords.ts

```diff
@@ -1,33 +1,55 @@
-import { MessageEmbed } from 'discord.js';
+import { SlashCommandSubcommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, MessageEmbed } from 'discord.js';
 import { fetchGroupDetails, fetchGroupRecords } from '../../../api/modules/groups';
 import { GroupRecordEntry } from '../../../api/types';
 import config from '../../../config';
-import { Command, ParsedMessage } from '../../../types';
-import { getAbbreviation, getEmoji, getMetricName, toKMB } from '../../../utils';
+import { getServer } from '../../../database/services/server';
+import { SubCommand } from '../../../types';
+import { getEmoji, getMetricName, toKMB } from '../../../utils';
 import CommandError from '../../CommandError';
 
-class GroupRecords implements Command {
-  name: string;
-  template: string;
+class GroupRecords implements SubCommand {
   requiresGroup?: boolean | undefined;
+  slashCommand?: SlashCommandSubcommandBuilder;
+  subcommand?: boolean | undefined;
 
   constructor() {
-    this.name = 'View group records';
-    this.template = '!group records {metric}? [--6h/--day/--week/--month/--year]';
     this.requiresGroup = true;
+    this.slashCommand = new SlashCommandSubcommandBuilder()
+      .addStringOption(option =>
+        option
+          .setName('metric')
+          .setDescription('The category to show records for')
+          .setAutocomplete(true)
+          .setRequired(true)
+      )
+      .addStringOption(option =>
+        option
+          .setName('period')
+          .setDescription('The period to show records for')
+          .addChoices([
+            ['5 Min', '5min'],
+            ['Day', 'day'],
+            ['Week', 'week'],
+            ['Month', 'month'],
+            ['Year', 'year']
+          ])
+          .setRequired(true)
+      )
+      .setName('records')
+      .setDescription('View group records');
+    this.subcommand = true;
   }
 
-  activated(message: ParsedMessage) {
-    const { command, args } = message;
-    return command === 'group' && args.length >= 1 && args[0] === 'records';
-  }
-
-  async execute(message: ParsedMessage) {
-    const groupId = message.originServer?.groupId || -1;
-    const metric = this.getMetricArg(message.args);
-    const period = this.getPeriodArg(message.args);
+  async execute(message: CommandInteraction) {
+    const guildId = message.guild?.id;
+    const server = await getServer(guildId); // maybe cache it so we don't have to do this
+    const groupId = server?.groupId || -1;
+    const metric = message.options.getString('metric', true);
+    const period = message.options.getString('period', true);
 
     try {
+      message.deferReply();
```

Comment:
race condition: If you don't `await` this and end up replying before this one is sent, you'll get an error.
It's unlikely as both answers are on discord, but I've seen it happen, so better `await` all of the `deferReply`